test/jest/mock-store.js

```diff
@@ -0,0 +1,69 @@
+import { MAINNET_CHAIN_ID } from '../../shared/constants/network';
+
+export const createSwapsMockStore = () => {
```

Comment:
I like having a mock store for tests! Do we think there is value in creating specific store mocks (swaps, vs general state?) I think instead of separating by feature it might make sense to separate by a scenario where it makes sense. Like "mainnetAcctWithMultipleTokens" or "customNetworkPowerUser" -- this is just brainstorming atm! 

also related but unrelated: #10895 introduces a localization "mock" to the provider

Reply:
Nice, will check the localization "mock", so we can get translations into our unit tests instead of just keys!

About creating a mock store for a feature vs one general state.. I think a general state could be pretty big if it would contain data for all features. I would do it per feature and if there is some shared state across features, it can be abstracted in the mock-store.js file into its own function and used in feature specific functions like `createSwapsMockStore`. For example:
```
export const createSwapsMockStore = () => {
  ...createGeneralMockStore(),
  swapsKey1: 'swapsValue1',
  swapsKey2: 'swapsValue2',
}
```

For specific scenarios within a feature there is always an option to call e.g. `createSwapsMockStore` to get default feature state and then modify that state in the code. For example:
```
const swapsMockStore = createSwapsMockStore();
swapsMockStore.metamask.provider.chainId = BSC_CHAIN_ID;
swapsMockStore.metamask.selectedAddress: 'differentAddress0x0dcd5d886577d5081b0c52e242ef29e70be3e7bc'
```